### PR TITLE
Avoid init_p2p all the times

### PR DESCRIPTION
**Pull request by panyx0718**
Fix https://github.com/PaddlePaddle/Paddle/issues/9665

### CODE REVIEW

File: paddle/fluid/framework/init.cc [hunk 1]
```diff
@@ -64,7 +64,7 @@ void InitP2P(int count) {
 #endif
 }
 
-void InitDevices() {
+void InitDevices(bool init_p2p) {
```
Comment (wangkuiyi) [inline paddle/fluid/framework/init.cc]: It seems that `InitDevice` is always called with `init_p2p=true`. Do we plan to change some of these invocations to be with false?

Comment (panyx0718) [reply paddle/fluid/framework/init.cc]: see __init__.py, For all tests, it should be false

Comment (panyx0718) [reply paddle/fluid/framework/init.cc]: https://github.com/PaddlePaddle/Paddle/pull/9699/files#diff-d4bed27d98cf7849623c79d9d2d3c32bR114

Comment (wangkuiyi) [reply paddle/fluid/framework/init.cc]: I see. Cool!

**Comment by Xreki** [comment]
Can we set the default value of `init_p2p`, because in users' inference codes, they need to call `InitDevices`. This change may fail their old codes if they update the library.

**Comment by panyx0718** [comment]
@Xreki if there are more than a handful of users, we can set the default. Otherwise, we can just ask them to change their codes.

**Pull request merged**